# VEuPathDB Data Report
Reports for the data fields in the API. 


In [1]:
import requests
from IPython.display import display 
import requests 
import json
import pandas as pd
pd.set_option('display.max_rows', None)
pd.set_option('display.max_colwidth', None)

 

In [2]:

"""
Helper Functions 
"""
def http_request(req):
    print("[INFO] status code response:",req.status_code)
    print("[INFO] reason:",req.reason)
    pretty_json=json.loads(req.text)
    print("[INFO] text response:", json.dumps(pretty_json, indent=4))


Run the request below to gather the datasets. 

In [3]:
# all datasets Search
all_datasets='https://veupathdb.org/veupathdb/service/record-types/dataset/searches/AllDatasets/reports/standard?reportConfig={"attributes":["primary_key","organism_prefix","project_id","eupath_release","newcategory","summary","contact","wdk_weight","version","institution","build_number_introduced","pmids_download","release_policy","short_attribution","type","genecount"],"tables":["Publications","Contacts","GenomeHistory","DatasetHistory","Version","References","HyperLinks","GeneTypeCounts","TranscriptTypeCounts"],"attributeFormat":"text"}'

all_req=requests.get(all_datasets)
datasets_dict=all_req.json()


In [4]:
records_list=datasets_dict['records']

In [6]:
# example record
#print(json.dumps(records_list[0], indent=4))

In [5]:
def setup_df(_data, key, verbose=True):
    _df = pd.DataFrame(_data, index=[0])
    #_df.columns = ["key", "value"]

    if key ==  'tables.GenomeHistory': tables_gh.append(_df)
    elif key ==  'tables.Publications': tables_pubs.append(_df)
    elif key ==  'tables.Contacts': tables_contacts.append(_df)
    elif key ==  'tables.Version': tables_version.append(_df)
    elif key ==  'tables.GeneTypeCounts': tables_gtc.append(_df)
    elif key ==  'tables.References': tables_ref.append(_df)
    elif key ==  'tables.DatasetHistory': tables_dh.append(_df)
    elif key ==  'tables.HyperLinks': tables_hl.append(_df)
    elif key ==  'tables.TranscriptTypeCounts': tables_ttc.append(_df)
    
    
    if verbose == True:
        display(_df)
    return _df



tables_gh=[]
tables_pubs=[]
tables_contacts=[]
tables_version=[]
tables_gtc=[]
tables_ref=[]
tables_dh=[]
tables_hl=[]
tables_ttc=[]

attributes=[]

for _dict in records_list:

    _id=_dict['id'][0]['value']
    #print('[INFO] dataset id: ', _id)
    verbose=False
    if verbose==True:
        print("[INFO] keys: ", _dict.keys())
        print("[INFO] displayName: ", _dict['displayName'])
        print("[INFO] recordClassName: ", _dict['recordClassName'])
        print("[INFO] id: ", _dict['id'])
        print("[INFO] tableErrors: ", _dict['tableErrors'])

    attributes_df=pd.DataFrame(_dict['attributes'], index=[0])
    attributes.append(attributes_df)

    ############################    
    #  get available tables 
    ############################
    for key in list(_dict['tables'].keys()):
        # assign variables
        _key="tables."+key
        #print("elif key ==  '%s': tables_.append(_df)"%_key )
        temp_list=_dict['tables'][key]

        
        if len(temp_list) == 1:
            _data=temp_list[0]
            if type(_data) is dict: 
                setup_df(_data, _key, verbose=False)
            else:
                print(_data)

        elif len(temp_list) > 1:
            for _data in temp_list:
                if type(_data) is dict:
                    setup_df(_data, _key, verbose=False)
                else:
                    print(_data)

        elif len(temp_list) == 0:
            #print("[INFO] key: %s | val: %s "%(key, temp_list))
            pass

  
attr_df=pd.concat(attributes)
#display(attr_df.head())

df_gh=pd.concat(tables_gh)
#display(df_gh.tail())

df_pubs=pd.concat(tables_pubs)
#display(df_pubs.tail())

df_contacts=pd.concat(tables_contacts)
#display(df_contacts.tail())

df_version=pd.concat(tables_version)
#display(df_version.tail())

df_gtc=pd.concat(tables_gtc)
#display(df_gtc.tail())

df_ref=pd.concat(tables_ref)
#display(df_ref.tail())

df_dh=pd.concat(tables_dh)
#display(df_dh.tail())

df_hl=pd.concat(tables_hl)
#display(df_hl.tail())

df_ttc=pd.concat(tables_ttc)
#display(df_ttc.tail())




## `records.attributes`

In [12]:
attr_df.head()

summary  \
0                                                                                                                          Genome Sequence for <i>Epichloe glyceriae</i> E277   
0                                                                                                                DNA damage-induced transcriptome changes in Candida glabrata   
0                     Automatic mosquito surveillance from 2017-06-21 to 2017-12-31 by BG-Counter smart traps owned by Collier Mosquito Control District. (MapVEu VBP0000343)   
0  characterization of genetic differences between insecticide resistant and susceptible Culex quinquefasciatus mosquitoes using targeted DNA sequencing. (MapVEu VBP0000454)   
0                                                                            <i>G. lamblia</i> Assemblage A Proteomics of virulent and avirulent Strains (Mass Spec peptides)   

  wdk_weight build_number_introduced  \
0          0                      51   
0          0                      56   
0          0                      47   
0          0                      47   
0          0                      25   

                              organism_prefix                   type  \
0          <i>Epichloe glyceriae</i> E277<br>                 genome   
0         <i>Candida glabrata</i> CBS 138<br>  transcript_expression   
0                                        <br>     population_biology   
0                                        <br>     population_biology   
0  <i>Giardia Assemblage</i> A isolate WB<br>     protein_expression   

                                                                                                                                                                         primary_key  \
0                                                                                                                                     <i>Epichloe glyceriae</i> E277 Genome Sequence   
0                                                                                                                       DNA damage-induced transcriptome changes in Candida glabrata   
0                                                                                              Mosquito surveillance by smart traps owned by Collier Mosquito Control District. 2017   
0  Using targeted next-generation sequencing to characterize genetic differences associated with insecticide resistance in Culex quinquefasciatus populations from the southern U.S.   
0                                                                                                                                 Virulent vs Avirulent Trophozoites (B-2041, H-106)   

      version release_policy         newcategory  \
0  2011-09-02           None             Genomes   
0  2020-12-15           None              RNASeq   
0  2018-10-17           None  Population Biology   
0  2019-05-22           None  Population Biology   
0  2014-11-12           None  Protein expression   

                                 pmids_download  \
0  https://www.ncbi.nlm.nih.gov/pubmed/23468653   
0  https://www.ncbi.nlm.nih.gov/pubmed/33323516   
0          https://www.ncbi.nlm.nih.gov/pubmed/   
0          https://www.ncbi.nlm.nih.gov/pubmed/   
0  https://www.ncbi.nlm.nih.gov/pubmed/25266764   

                                                                                                                       institution  \
0                                                                       Department of Plant Pathology, University of Kentucky, USA   
0                                                                  Center for Discovery and Innovation, Hackensack Meridian Health   
0                                                                                                Collier Mosquito Control District   
0  Division of Vector-Borne Diseases, Centers for Disease Control and Prevention, Fort Collins, Colorado, United States of America   
0                                 

In [15]:
attr_df['organism_prefix'].head()

0            <i>Epichloe glyceriae</i> E277<br>
0           <i>Candida glabrata</i> CBS 138<br>
0                                          <br>
0                                          <br>
0    <i>Giardia Assemblage</i> A isolate WB<br>
Name: organism_prefix, dtype: object

In [9]:
attr_df[['type', 'newcategory']].tail(10)

type           newcategory
0  transcript_expression  DNA Microarray Assay
0                 genome               Genomes
0     population_biology    Population Biology
0  transcript_expression                RNASeq
0     protein_expression    Protein expression
0  transcript_expression                RNASeq
0     population_biology    Population Biology
0  transcript_expression                RNASeq
0               isolates     Genetic variation
0                 genome               Genomes

## `records.tables`

In [14]:
display(df_version['organism'].head())

0            Epichloe glyceriae E277
0           Candida glabrata CBS 138
0    Giardia Assemblage A isolate WB
0            Aedes albopictus Foshan
0             Aedes aegypti LVP_AGWG
Name: organism, dtype: object

In [18]:
import numpy as np

In [20]:
display(df_contacts['affiliation'].head())
display(attr_df['institution'].head())


0                                                                         Department of Plant Pathology, University of Kentucky, USA
0                                                                    Center for Discovery and Innovation, Hackensack Meridian Health
0                                                                    Center for Discovery and Innovation, Hackensack Meridian Health
0                                                                                                  Collier Mosquito Control District
0    Division of Vector-Borne Diseases, Centers for Disease Control and Prevention, Fort Collins, Colorado, United States of America
Name: affiliation, dtype: object

0                                                                         Department of Plant Pathology, University of Kentucky, USA
0                                                                    Center for Discovery and Innovation, Hackensack Meridian Health
0                                                                                                  Collier Mosquito Control District
0    Division of Vector-Borne Diseases, Centers for Disease Control and Prevention, Fort Collins, Colorado, United States of America
0                                                                                Walter and Eliza Hall Institute of Medical Research
Name: institution, dtype: object

False

In [11]:
 df_gtc.head()

gene_count            gene_type     dataset_id
0        337           ncRNA gene  DS_01b2dc5b05
0       4886  protein coding gene  DS_01b2dc5b05
0          9           pseudogene  DS_01b2dc5b05
0        173           ncRNA gene  DS_01c6cee55d
0       9869  protein coding gene  DS_01c6cee55d

In [12]:
attr_df.tail()

summary wdk_weight  \
0  Mouse transcriptome time course during infecti...          0   
0  In 2011, Benin&#8217;s National Malaria Contro...          0   
0  Transcriptome of <i>T. cruzi</i> Y strain duri...          0   
0  Whole-genome sequencing (WGS) was performed fo...          0   
0  Genome Sequence and Annotation of <i>Hansenias...          0   

  build_number_introduced                                    organism_prefix  \
0                      53  <i>Crithidia fasciculata</i> strain Cf-Cl<br><...   
0                      47                                               <br>   
0                      34      <i>Trypanosoma cruzi</i> strain CL Brener<br>   
0                      38             <i>Aspergillus fumigatus</i> Af293<br>   
0                      52  <i>Hanseniaspora guilliermondii</i> strain UTA...   

                    type                                        primary_key  \
0  transcript_expression  Mouse transcriptome during P. vinckei CY infec...   
0     population_biology  Resistance status of A. gambiae s.l. following...   
0  transcript_expression  Transcriptome during Infection in Human Fibrob...   
0               isolates  Genomic Context of Azole-Resistance Mutations ...   
0                 genome  <i>Hanseniaspora guilliermondii</i> strain UTA...   

      version release_policy         newcategory  \
0  2019-01-25           None              RNASeq   
0  2020-10-15           None  Population Biology   
0  2016-04-05           None              RNASeq   
0  2015-04-24           None   Genetic variation   
0  2017-02-02           None             Genomes   

                                 pmids_download  \
0  https://www.ncbi.nlm.nih.gov/pubmed/30683099   
0  https://www.ncbi.nlm.nih.gov/pubmed/31941494   
0  https://www.ncbi.nlm.nih.gov/pubmed/27046031   
0  https://www.ncbi.nlm.nih.gov/pubmed/26037120   
0  https://www.ncbi.nlm.nih.gov/pubmed/30462193   

                                         institution  project_id  \
0  King Abdullah University of Science and Techno...      HostDB   
0  Centre de Recherche Entomologique de Cotonou (...  VectorBase   
0          Harvard T.H. Chan School of Public Health   TriTrypDB   
0                            Imperial College London     FungiDB   
0  Department of Bioengineering, Institute of Bio...     FungiDB   

  short_attribution           contact genecount                 eupath_release  
0  Ramaprasad et al        Arnab Pain      None  EuPathDB rel. 53, 2021-JUL-21  
0       Arthur Sovi       Arthur Sovi      None  EuPathDB rel. 47, 2020-APR-07  
0         Li et al.  Barbara Burleigh      None  EuPathDB rel. 34, 2017-SEP-07  
0    Matthew Fisher    Matthew Fisher      None  EuPathDB rel. 38, 2018-JUL-05  
0         Nuno Mira         Nuno Mira      4152  EuPathDB rel. 52, 2021-MAY-20

In [14]:
df_gh_=df_gh.set_index("dataset_id")

In [15]:
df_gh_.head(50)

note  \
dataset_id                                                         
DS_0011425ac1                                               None   
DS_0024327063                                               None   
DS_002a5dad83                                               None   
DS_00672484d6                                               None   
DS_006cb7d943                                               None   
DS_00893e0bf8                                               None   
DS_00b47a520a                                               None   
DS_011c12b8d3                                               None   
DS_01225327f4                                               None   
DS_012298da2c                                               None   
DS_019172d4ff                                               None   
DS_01b2dc5b05                 rebuild with INSDC accession added   
DS_01b2dc5b05                                               None   
DS_01b2dc5b05                  re-annotated, data source changed   
DS_01c6cee55d                                    newly-annotated   
DS_01c6cee55d  consistent date format and INSDC accession whe...   
DS_01c6cee55d                                               None   
DS_01d08f2525                                               None   
DS_022384d18a                                               None   
DS_022384d18a                 rebuild with INSDC accession added   
DS_0226b5025b                                               None   
DS_02354d7d63                                               None   
DS_02354d7d63                 rebuild with INSDC accession added   
DS_0241addc8e                                               None   
DS_02717866d1                                               None   
DS_0273077ad5                                               None   
DS_02c0ef3cc2                                               None   
DS_02caff887d                                               None   
DS_03395fd8e3                                               None   
DS_0339e44248                                               None   
DS_033b09f437             functional annotation updated from CGD   
DS_033b09f437                 rebuild with INSDC accession added   
DS_033b09f437                                               None   
DS_033cf9c74d                                               None   
DS_033e62f4ec                                               None   
DS_035232648f                                               None   
DS_0360a7ae66                                               None   
DS_03824cf332                                               None   
DS_03986f8069                                               None   
DS_0399018270                 rebuild with INSDC accession added   
DS_0399018270                                               None   
DS_0399018270                  re-annotated, data source changed   
DS_03a864a884                                               None   
DS_03aef8200c                                               None   
DS_03d8e25e02                                               None   
DS_03f94e0879                                               None   
DS_04102dc2b8                                               None   
DS_0445efcab5                                               None   
DS_04464ff60b                                               None   
DS_04666bdd43                                               None   

              annotation_version annotation_source release_number build  \
dataset_id                                                                
DS_0011425ac1               None              None             51    51   
DS_0024327063               None              None             56    56   
DS_002a5dad83               None              None             47    47   
DS_00672484d6               None              None             47    47   
DS_006cb7d943               None              None             25    25  

## `tables.Publications`

In [6]:
df_pubs.head()

pubmed_link  \
0  http://www.ncbi.nlm.nih.gov/pubmed/23468653   
0  http://www.ncbi.nlm.nih.gov/pubmed/33323516   
0  http://www.ncbi.nlm.nih.gov/pubmed/25266764   
0  http://www.ncbi.nlm.nih.gov/pubmed/33553008   
0  http://www.ncbi.nlm.nih.gov/pubmed/25851262   

                                                                                                                                                                                                                                                                                                                                                                      citation  \
0                                                                                                                                                                                                    Plant-symbiotic fungi as chemical engineers: multi-genome analysis of the clavicipitaceae reveals dynamics of alkaloid loci. Schardl et al. PLoS Genet 2013;9(2):e1003323   
0                                                        A Noncanonical DNA Damage Checkpoint Response in a Major Fungal Pathogen. Shor et al. mBio 2020 12 15;11(6):<?xml version="1.0" ?> <!DOCTYPE PubmedArticleSet PUBLIC "-//NLM//DTD PubMedArticle, 1st January 2019//EN" "https://dtd.nlm.nih.gov/ncbi/pubmed/out/pubmed_190101.dtd"> <PubmedArticleSet><PubmedArticle>   
0                                                                                                                                                                                                                       Proteomic analysis in Giardia duodenalis yields insights into strain virulence and antigenic variation. Emery et al. Proteomics 2014;14(21-22):2523-34   
0  High-Resolution Mapping of Transcription Initiation in the Asexual Stages of <i>Toxoplasma gondii</i>. Markus et al. Front Cell Infect Microbiol 2020;10(<?xml version="1.0" ?> <!DOCTYPE PubmedArticleSet PUBLIC "-//NLM//DTD PubMedArticle, 1st January 2019//EN" "https://dtd.nlm.nih.gov/ncbi/pubmed/out/pubmed_190101.dtd"> <PubmedArticleSet> <PubmedArticle>):617998   
0                                                                                                                                                     Genome sequence comparison of Aspergillus fumigatus strains isolated from patients with pulmonary aspergilloma and chronic necrotizing pulmonary aspergillosis. Takahashi-Nakaguchi et al. Med. Mycol. 2015;53(4):353-60   

      dataset_id      pmid                                          url  
0  DS_0011425ac1  23468653  http://www.ncbi.nlm.nih.gov/pubmed/23468653  
0  DS_0024327063  33323516  http://www.ncbi.nlm.nih.gov/pubmed/33323516  
0  DS_006cb7d943  25266764  http://www.ncbi.nlm.nih.gov/pubmed/25266764  
0  DS_011c12b8d3  33553008  http://www.ncbi.nlm.nih.gov/pubmed/33553008  
0  DS_012298da2c  25851262  http://www.ncbi.nlm.nih.gov/pubmed/25851262

## `tables.contacts`

In [7]:
df_contacts.head()

contact_name  \
0  Christopher Schardl   
0           Erika Shor   
0         David Perlin   
0          Keira Lucas   
0        Linda Kothera   

                                                                                                                       affiliation  \
0                                                                       Department of Plant Pathology, University of Kentucky, USA   
0                                                                  Center for Discovery and Innovation, Hackensack Meridian Health   
0                                                                  Center for Discovery and Innovation, Hackensack Meridian Health   
0                                                                                                Collier Mosquito Control District   
0  Division of Vector-Borne Diseases, Centers for Disease Control and Prevention, Fort Collins, Colorado, United States of America   

      dataset_id                                 dataset_name  
0  DS_0011425ac1                 eglyE277_primary_genome_RSRC  
0  DS_0024327063  cglaCBS138_Shor_2020_Cglabr_ebi_rnaSeq_RSRC  
0  DS_0024327063  cglaCBS138_Shor_2020_Cglabr_ebi_rnaSeq_RSRC  
0  DS_002a5dad83                 PopBio_Study_VBP0000343_RSRC  
0  DS_00672484d6                 PopBio_Study_VBP0000454_RSRC

## `tables.Version`

In [8]:
df_version.head()

organism  project_id     dataset_id  \
0          Epichloe glyceriae E277     FungiDB  DS_0011425ac1   
0         Candida glabrata CBS 138     FungiDB  DS_0024327063   
0  Giardia Assemblage A isolate WB   GiardiaDB  DS_006cb7d943   
0          Aedes albopictus Foshan  VectorBase  DS_00893e0bf8   
0           Aedes aegypti LVP_AGWG  VectorBase  DS_00b47a520a   

                                               dataset_name     version  
0                              eglyE277_primary_genome_RSRC  2011-09-02  
0               cglaCBS138_Shor_2020_Cglabr_ebi_rnaSeq_RSRC  2020-12-15  
0          gassAWB_massSpec_Emery_Virulence_Proteomics_RSRC  2014-11-12  
0                      aalbFoshan_SRP052205_ebi_rnaSeq_RSRC     01-2020  
0  aaegLVP_AGWG_microarrayPlatform_A-MEXP-1878-Agilent_RSRC  2014-05-02

## `tables.GeneTypeCounts`

In [9]:
df_gtc.head()

gene_count            gene_type     dataset_id
0        337           ncRNA gene  DS_01b2dc5b05
0       4886  protein coding gene  DS_01b2dc5b05
0          9           pseudogene  DS_01b2dc5b05
0        173           ncRNA gene  DS_01c6cee55d
0       9869  protein coding gene  DS_01c6cee55d

## `tables.DatasetHistory`  

In [10]:
df_dh.head()

note release_number build release_date     dataset_id     version
0  None             51    51   2021-03-16  DS_0011425ac1  2011-09-02
0  None             56    56   2022-02-15  DS_0024327063  2020-12-15
0  None             47    47   2020-04-07  DS_002a5dad83  2018-10-17
0  None             47    47   2020-04-07  DS_00672484d6  2019-05-22
0  None             25    25   2015-07-23  DS_006cb7d943  2014-11-12

## `tables.HyperLinks`

In [11]:
df_hl.head()

hyper_link     dataset_id  \
0  https://www.ncbi.nlm.nih.gov/assembly/GCA_000225285.2  DS_0011425ac1   
0     https://www.ncbi.nlm.nih.gov/bioproject/PRJNA67247  DS_0011425ac1   
0    https://www.ncbi.nlm.nih.gov/bioproject/PRJNA655241  DS_0024327063   
0             http://proteomecentral.proteomexchange.org  DS_006cb7d943   
0        http://www.ncbi.nlm.nih.gov/sra/?term=SRP052205  DS_00893e0bf8   

                                                          text  \
0                                        GenBank Assembly page   
0                                              NCBI Bioproject   
0                                              NCBI BioProject   
0  Data located at Proteome Central under identifier PXD000452   
0                                               SRA project id   

                                                     url  
0  https://www.ncbi.nlm.nih.gov/assembly/GCA_000225285.2  
0     https://www.ncbi.nlm.nih.gov/bioproject/PRJNA67247  
0    https://www.ncbi.nlm.nih.gov/bioproject/PRJNA655241  
0             http://proteomecentral.proteomexchange.org  
0        http://www.ncbi.nlm.nih.gov/sra/?term=SRP052205

## `tables.TranscriptTypeCounts`  


In [12]:
df_ttc.head()

gene_type transcript_count     dataset_id         transcript_type
0           ncRNA gene               15  DS_01b2dc5b05                    rRNA
0           ncRNA gene              322  DS_01b2dc5b05                    tRNA
0  protein coding gene             4886  DS_01b2dc5b05                    mRNA
0           pseudogene                9  DS_01b2dc5b05  pseudogenic_transcript
0           ncRNA gene                2  DS_01c6cee55d                    rRNA

---

In [204]:

# data report aggregations 
def data_report(_df):
    print("\n[INFO] data example:")
    display(_df.tail())

    print("\n[INFO] info(): ")
    display(_df.info())

    print("\n[INFO] describe() ")
    display(_df.describe().T)
    
    print("\n[INFO] aggregations:")
    display(_df.aggregate(['min', 'max']))

    #print("\n[INFO] :")
    #for key in _df.columns.values:
     # if _df[key].dtype == object:
          #_df[key] = _df[key].astype('category')
          #display(_df.quantile(axis=)) #axis=1 for col-wise

## Tables

#### `GenomeHistory` Table

In [165]:
# data sample
display(df_gh.tail(15))

note annotation_source annotation_version  \
0                                None              None               None   
0                                None              None               None   
0                                None              None               None   
0                                None              None               None   
0                                None              None               None   
0                                None           GenBank       Mar 21, 2015   
0  rebuild with INSDC accession added           GenBank       Mar 21, 2015   
0                                None              None               None   
0                                None              None               None   
0                                None              None               None   
0                                None              None               None   
0                                None              None               None   
0                                None              None               None   
0                                None              None               None   
0                                None           GenBank       Feb 02, 2017   

  release_number   genome_version build release_date genome_source  \
0             48             None    48   2020-08-27          None   
0             22             None    22   2014-09-10          None   
0             53             None    53   2021-07-21          None   
0             29             None    29   2016-10-12          None   
0             49             None    49   2020-11-05          None   
0             46     Mar 21, 2015    46   2019-11-06       GenBank   
0             51  GCA_000365145.2    51   2021-03-16       GenBank   
0             47             None    47   2020-04-07          None   
0             47             None    47   2020-04-07          None   
0             28             None    28   2016-03-31          None   
0             53             None    53   2021-07-21          None   
0             47             None    47   2020-04-07          None   
0             34             None    34   2017-09-07          None   
0             38             None    38   2018-07-05          None   
0             52  GCA_900119595.1    52   2021-05-20         INSDC   

      dataset_id functional_annotation_source functional_annotation_version  
0  DS_ffbe968d03                         None                          None  
0  DS_ffbe968d03                         None                          None  
0  DS_ffbe968d03                         None                          None  
0  DS_ffbe968d03                         None                          None  
0  DS_ffbe968d03                         None                          None  
0  DS_ffbe968d03                         None                          None  
0  DS_ffbe968d03                         None                          None  
0  DS_ffbe968d03                         None                          None  
0  DS_ffbe968d03                         None                          None  
0  DS_ffbe968d03                         None                          None  
0  DS_ffbe968d03                         None                          None  
0  DS_ffbe968d03                         None                          None  
0  DS_ffbe968d03                         None                          None  
0  DS_ffbe968d03                         None                          None  
0  DS_ffbe968d03                         None                          None

In [123]:
df_gh.shape

(3377, 11)

In [124]:
# quick info stats 
display(df_gh.info())

<class 'pandas.core.frame.DataFrame'>
Int64Index: 3377 entries, 0 to 0
Data columns (total 11 columns):
 #   Column                         Non-Null Count  Dtype 
---  ------                         --------------  ----- 
 0   note                           960 non-null    object
 1   annotation_version             1513 non-null   object
 2   annotation_source              1513 non-null   object
 3   release_number                 3377 non-null   object
 4   build                          3377 non-null   object
 5   release_date                   3377 non-null   object
 6   genome_version                 1558 non-null   object
 7   genome_source                  1558 non-null   object
 8   dataset_id                     3377 non-null   object
 9   functional_annotation_source   325 non-null    object
 10  functional_annotation_version  325 non-null    object
dtypes: object(11)
memory usage: 316.6+ KB


None

In [158]:
# describe data
display(df_gh.describe())

note annotation_source  \
count                                  960              1513   
unique                                 112                61   
top     rebuild with INSDC accession added           GenBank   
freq                                   276               638   

       annotation_version release_number genome_version build release_date  \
count                1513           3377           1558  3377         3377   
unique                482             45            950    45           45   
top                   n/a             47   May 28, 2016    47   2020-04-07   
freq                  128           1096             42  1096         1096   

       genome_source     dataset_id functional_annotation_source  \
count           1558           3377                          325   
unique            53              1                            3   
top          GenBank  DS_ffbe968d03                       GeneDB   
freq             814           3377                          264   

       functional_annotation_version  
count                            325  
unique                            35  
top                     May 06, 2020  
freq                              30



### `records.tables.GenomeHistory`   
  
 "Is this a reference genome that this dataset is based on? If so, I recommend capturing with `isBasedOn`"
- No, looking closer it seems to be developer notes, and only has 960 non-null values, (out of 3377 records)

In [136]:
# inspect notes 
mask=df_gh.note.notnull()
note_series=df_gh.note.where(mask).dropna()

In [137]:
display(note_series[:10])

0                   rebuild with INSDC accession added
0                    re-annotated, data source changed
0                                      newly-annotated
0    consistent date format and INSDC accession whe...
0                   rebuild with INSDC accession added
0                   rebuild with INSDC accession added
0               functional annotation updated from CGD
0                   rebuild with INSDC accession added
0                   rebuild with INSDC accession added
0                    re-annotated, data source changed
Name: note, dtype: object

`records.tables.GenomeHistory.annotation_source`   
`records.tables.GenomeHistory.annotation_version`  

In [161]:
mask=df_gh[['annotation_source', 'annotation_version']].notnull()
annot_df=df_gh[['annotation_source', 'annotation_version']].where(mask).dropna()
annot_df.tail()

annotation_source annotation_version
0      SchnauferLab       Jun 03, 2014
0      SchnauferLab       Jun 13, 2013
0           GenBank       Mar 21, 2015
0           GenBank       Mar 21, 2015
0           GenBank       Feb 02, 2017

`records.tables.GenomeHistory.release_number`    
`records.tables.GenomeHistory.release_date`

In [164]:
display(df_gh.filter(regex="release").tail())


release_number release_date
0             53   2021-07-21
0             47   2020-04-07
0             34   2017-09-07
0             38   2018-07-05
0             52   2021-05-20

In [144]:
# versions  -- annotation_version, genome_version, functional_annotation_version
version_df= df_gh.filter(regex="version")
vr_mask=version_df.notnull()
vr_df=version_df.where(mask).dropna()

print("\n[INFO] number of unique values: ")
display(vr_df.nunique())

print("\n[INFO] total non-null values: ")
display(vr_df.count())



[INFO] number of unique values: 


annotation_version               36
genome_version                   70
functional_annotation_version    35
dtype: int64


[INFO] total non-null values: 


annotation_version               323
genome_version                   323
functional_annotation_version    323
dtype: int64

In [187]:
vr_df.head(15)

annotation_version   genome_version functional_annotation_version
0       Sep 27, 2015  GCA_000002655.1                  Sep 01, 2020
0       Sep 27, 2015     Sep 27, 2015                  May 04, 2018
0       Sep 27, 2015  GCA_000002655.1                  Oct 01, 2021
0       Sep 27, 2015     Sep 27, 2015                  Nov 21, 2017
0       Sep 27, 2015     Sep 27, 2015                  May 20, 2017
0       Sep 27, 2015     Sep 27, 2015                  Sep 01, 2020
0       Sep 27, 2015     Sep 27, 2015                  May 06, 2020
0       Sep 27, 2015     Sep 27, 2015                  Jul 26, 2019
0       Sep 27, 2015     Sep 27, 2015                  Jan 15, 2019
0       Sep 27, 2015     Sep 27, 2015                  Feb 16, 2017
0       Oct 04, 2015     Oct 04, 2015                  Jan 15, 2019
0       Oct 04, 2015  GCA_000184455.3                  Jan 15, 2019
0       Dec 07, 2015     Dec 07, 2015                  Oct 22, 2018
0       Dec 07, 2015     Dec 07, 2015                  Jan 15, 2019
0       Dec 07, 2015     Dec 07, 2015                  Mar 30, 2019

In [166]:
# non-null large dataset
nonnull_mask=df_gh.notnull()
nonnull_df=df_gh.where(nonnull_mask).dropna()

In [172]:
nonnull_df['annotation_source'].unique(), nonnull_df['genome_source'].unique(), nonnull_df['functional_annotation_source'].unique()

(array(['AspGD', 'GeneDB', 'n/a', 'GenBank', 'RequenaLab', 'Broad',
        'HammondKosackLab', 'Sanger'], dtype=object),
 array(['GenBank', 'AspGD', 'GeneDB', 'RequenaLab', 'Broad',
        'HammondKosackLab'], dtype=object),
 array(['Sanger', 'GeneDB', 'GenBank'], dtype=object))

In [184]:
df_gh.loc[df_gh['dataset_id'] == "DS_ffbe968d03"].head()


note annotation_source annotation_version release_number   genome_version  \
0  None              None               None             51  GCA_000225285.2   
0  None              None               None             56             None   
0  None              None               None             47             None   
0  None              None               None             47             None   
0  None              None               None             25             None   

  build release_date genome_source     dataset_id  \
0    51   2021-03-16         INSDC  DS_ffbe968d03   
0    56   2022-02-15          None  DS_ffbe968d03   
0    47   2020-04-07          None  DS_ffbe968d03   
0    47   2020-04-07          None  DS_ffbe968d03   
0    25   2015-07-23          None  DS_ffbe968d03   

  functional_annotation_source functional_annotation_version  
0                         None                          None  
0                         None                          None  
0                         None                          None  
0                         None                          None  
0                         None                          None

---